introduction/edge-functions.mdx

```diff
@@ -0,0 +1,96 @@
+1. Under "Development", select "Edge Function", then click the "Create" button.
+1. Write and test your function.
+    1. Select the database you want to access and an API key if necessary
+    2. When you're finished, click deploy.
+
+#### Note:
+Functions should return a JSON-serializable object with a data field:
+```JavaScript
+return {
+  data: {
+    // your return object
+  }
+}
+```
+
+## Guides
+### Interacting with your Database
+Use the global `connection` object to access and manipulate your database.
+```JavaScript
+const customers = await connection.sql`SELECT * FROM customers;`
+return {
+  data: customers
+}
+```
+Select the database you would like to access from the "Database" dropdown, or select the database you want to use in your SQL query with the [USE command](https://docs.sqlitecloud.io/docs/commands/use-database).
+```JavaScript
+const customers = await connection.sql`USE DATABASE chinook.sqlite; SELECT * FROM customers;`;
+return {
+  data: customers
+}
+```
+
+### Storing and Accessing Environment Variables
+Environment variables can be stored in either the server settings file, database settings file, or stored directly in your database. 
+
+Use the following commands to set and read values in your server settings file:
+* [SET KEY](https://docs.sqlitecloud.io/docs/commands/set-key) <keyname> TO <keyvalue>
+* [GET KEY](https://docs.sqlitecloud.io/docs/commands/get-key) <keyname>
+* [REMOVE KEY](https://docs.sqlitecloud.io/docs/commands/remove-key) <keyname>
+* [LIST KEYS](https://docs.sqlitecloud.io/docs/commands/list-keys)
+
+Use the following commands to set and read values in your database-specific settings file:
+* [SET DATABASE](https://docs.sqlitecloud.io/docs/commands/set-database) <database_name> KEY <keyname> TO <keyvalue> 
+* [GET DATABASE](https://docs.sqlitecloud.io/docs/commands/get-database-key) <database_name> KEY <keyname> 
+* [REMOVE DATABASE](https://docs.sqlitecloud.io/docs/commands/remove-database-key) <database_name> KEY <keyname>
+* LIST DATABASE <database_name> KEYS
+
+### Handling Errors
+In case of error we return an HTTP error code and a JSON with the error message. Manually throwing an error in your code results in a 500 response.
+
+
+## Examples
+
+### Assigning and Notifying a Support Rep on User Sign up
+```JavaScript
+// Get secret from database
+const slackWebhookEndpoint = await connection.sql`GET KEY slack_webhook_endpoint`;
+
+// Get record sent in body via webhook
+const newCustomer = request.data
+
+// Sample business logic - assign and notify support rep
+// Get reps from database
+const reps = await connection.sql`SELECT id, name, country FROM reps`;
+const rep = reps.find(({ country }) => country === newCustomer.country.toLowerCase());
+
+// Define helpers to assign and notify
+const assignRep = async (repId, customerId) => await connection.sql`UPDATE customers SET SupportRepId = ${repId} WHERE CustomerId = ${customerId}`;
```

Comment:
I love the example

Reply:
In your example the `newCustomer` should become:

```
// Get record sent in body via webhook
const content = request.data;
const database = content.database;
const primaryKey = content.data[0];
const newCustomer = await connection.sql`USE DATABASE ${database}; SELECT * FROM customers WHERE CustomerId = ${primaryKey};`;
```

Reply:
Thank you for the clarification Tiziano!!!